src/seedu/addressbook/commands/CommandResult.java

```diff
@@ -36,13 +39,22 @@ public CommandResult(String feedbackToUser, List<? extends ReadOnlyPerson> relev
         this.relevantStatistics = null;
     }
 
-    public CommandResult(List<? extends Exam> relevantExams, String feedbackToUser,
+    public CommandResult(List<? extends ReadOnlyExam> relevantExams, String feedbackToUser) {
+        this.feedbackToUser = feedbackToUser;
+        this.relevantExams = relevantExams;
+        this.relevantStatistics = null;
```

Comment:
Seems a bit potato to assign so many fields to null though. Perhaps Optional might be better idea

Reply:
https://blog.joda.org/2014/11/optional-in-java-se-8.html. This portion of code will also be looked at again as later parts.